File: algorithms/ar-expsq/C/expsq.c
#include <stdio.h>

int expo(int base, int exp)
{
	int res = 1;
	while (exp > 0)
	{
		if (exp % 2 == 1)
			res = res * base;
		base = base * base;
		exp = exp / 2;
	}
	return res;
}
int main()
{
	int ans = expo(3, 3);
	printf("%d", ans);
}

File: algorithms/ar-expsq/Python3/expsq.py
def expo(base, exp):
    res=1
    while(exp>0):
        if(exp%2==1):
            res=res*base
        base=base*base
        exp=exp//2
    return res

x=int(input("Enter the base"))
y=int(input("Enter the exponent"))

print(x, "raised to power", y, "is",expo(x,y))
